FreeTubeApp/FreeTube
# Ctrl plus scroll to change playback rate

---
Ctrl plus scroll to change playback rate
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:

**Related issue**
Please link the issue your pull request is referring to. If this pull request fully resolves the relevant issue, put "closes" before the issue number. Example: "closes #123456".
closes #1713 
closes #616
closes #1238

**Description**
Please write a clear and concise description of what the pull request does.
- [x] Adds the ability to use the ctrl key and mouse wheel to quickly change the playback rate.
- [x] Changes the max playback rate from `3` to `8`
- [x] Adds the ability to use ctrl+click to return to the default playback rate.
- [x] Adds the ability to enable/disable the above in the `settings > player settings` screen (defaults to disabled)

**Screenshots (if appropriate)**
Please add before and after screenshots if there is a visible change.
Added Setting
![image](https://user-images.githubusercontent.com/1571418/135803925-0b244ce0-dbb1-4a48-83be-a8ce46fa11d8.png)
Added Tooltip for Setting
![image](https://user-images.githubusercontent.com/1571418/135803944-907060fe-41c6-4a18-a1e8-5c86ecb7b585.png)
Showing clicking without the control key pressed (the behavior is unchanged)
![9E06A815-9340-48D2-A1AD-6406B1463940](https://user-images.githubusercontent.com/1571418/135804532-2858b9c3-9046-4b08-9072-950c40bb8a09.GIF)
Showing Ctrl+Scrolling Up, Ctrl+Click, Ctrl+ Scrolling Down (NOTE: the key display software that I'm using does not support showing wheel events. You're just going to have to trust me.)
![A91EBEA4-5854-494B-87DD-7AB48BA255D2](https://user-images.githubusercontent.com/1571418/135804889-6e166528-9703-4e12-9e1e-fc502f2797fc.GIF)




**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
Please describe shortly how you tested it and whether there are any ramifications remaining. 

Manually tested:
* enabling/disabling the feature
  - No apparent user impact when disabled
  - Enabling the feature works as expected
* Scrolling up and down without holding ctrl
  - When "scroll to change volume" is turned on, it functions as expected
  - When "scroll to change volume" is turned off, it functions as expected
* Scrolling up and down while holding ctrl
  - When "scroll to change volume" is turned on, it functions as expected
  - When "scroll to change volume" is turned off, it functions as expected
* Ctrl+Click
  - This works as expected but there may be a better way to do this. See my first comment on this PR for more details

**Desktop (please complete the following information):**
  - OS: [e.g. iOS] : Windows 10 Home
  - OS Version: [e.g. 22]: Version 21H1, Build 19043.1237 with Windows Feature Experience Pack 120.2212.3530.0
  - FreeTube version: [e.g. 0.8] 0.14.0

**Additional context**
Add any other context about the problem here.
